From ef30f5f1f2df349c44ddc969801458c1685c57f7 Mon Sep 17 00:00:00 2001
From: Apaczer <94932128+Apaczer@users.noreply.github.com>
Date: Fri, 19 Dec 2025 20:52:29 +0100
Subject: [PATCH] config.def: update default HOTKEY configuration

+ increase Hotkey Enable Delay to 15 frames (otherwise interfere with in menu btn and combo buttons
---
 config.def.h          |  4 ++--
 config.def.keybinds.h | 36 ++++++++++++++++++------------------
 2 files changed, 20 insertions(+), 20 deletions(-)

diff --git a/config.def.h b/config.def.h
index 44db234e72..066ae1f57c 100644
--- a/config.def.h
+++ b/config.def.h
@@ -937,7 +937,7 @@
 #define DEFAULT_MENU_TOGGLE_GAMEPAD_COMBO INPUT_COMBO_NONE
 #endif
 
-#define DEFAULT_QUIT_GAMEPAD_COMBO INPUT_COMBO_NONE
+#define DEFAULT_QUIT_GAMEPAD_COMBO INPUT_COMBO_HOLD_START
 
 #if defined(VITA)
 #define DEFAULT_INPUT_BACKTOUCH_ENABLE false
@@ -1660,7 +1660,7 @@
 #define DEFAULT_INPUT_BIND_HOLD 0
 #endif
 #define DEFAULT_INPUT_POLL_TYPE_BEHAVIOR 2
-#define DEFAULT_INPUT_HOTKEY_BLOCK_DELAY 5
+#define DEFAULT_INPUT_HOTKEY_BLOCK_DELAY 15
 #define DEFAULT_INPUT_HOTKEY_DEVICE_MERGE false
 #define DEFAULT_INPUT_HOTKEY_FOLLOWS_PLAYER1 false
 
diff --git a/config.def.keybinds.h b/config.def.keybinds.h
index f1902ef99c..c7b7106cf7 100644
--- a/config.def.keybinds.h
+++ b/config.def.keybinds.h
@@ -953,8 +953,8 @@ static const struct retro_keybind retro_keybinds_1[] = {
    {
       NULL, NULL,
       AXIS_NONE, AXIS_NONE,
-      MENU_ENUM_LABEL_VALUE_INPUT_META_ENABLE_HOTKEY, RETROK_UNKNOWN,
-      RARCH_ENABLE_HOTKEY, NO_BTN, NO_BTN, 0,
+      MENU_ENUM_LABEL_VALUE_INPUT_META_ENABLE_HOTKEY, RETROK_ESCAPE,
+      RARCH_ENABLE_HOTKEY, NO_BTN, 8, 0,
       true
    },
    {
@@ -981,15 +981,15 @@ static const struct retro_keybind retro_keybinds_1[] = {
    {
       NULL, NULL,
       AXIS_NONE, AXIS_NONE,
-      MENU_ENUM_LABEL_VALUE_INPUT_META_RESET, RETROK_UNKNOWN,
-      RARCH_RESET, NO_BTN, NO_BTN, 0,
+      MENU_ENUM_LABEL_VALUE_INPUT_META_RESET, RETROK_TAB,
+      RARCH_RESET, NO_BTN, 4, 0,
       true
    },
    {
       NULL, NULL,
       AXIS_NONE, AXIS_NONE,
-      MENU_ENUM_LABEL_VALUE_INPUT_META_FAST_FORWARD_KEY, RETROK_UNKNOWN,
-      RARCH_FAST_FORWARD_KEY, NO_BTN, NO_BTN, 0,
+      MENU_ENUM_LABEL_VALUE_INPUT_META_FAST_FORWARD_KEY, RETROK_LSHIFT,
+      RARCH_FAST_FORWARD_KEY, NO_BTN, 0, 0,
       true
    },
    {
@@ -1016,15 +1016,15 @@ static const struct retro_keybind retro_keybinds_1[] = {
    {
       NULL, NULL,
       AXIS_NONE, AXIS_NONE,
-      MENU_ENUM_LABEL_VALUE_INPUT_META_REWIND, RETROK_UNKNOWN,
-      RARCH_REWIND, NO_BTN, NO_BTN, 0,
+      MENU_ENUM_LABEL_VALUE_INPUT_META_REWIND, RETROK_SPACE,
+      RARCH_REWIND, NO_BTN, 3, 0,
       true
    },
    {
       NULL, NULL,
       AXIS_NONE, AXIS_NONE,
-      MENU_ENUM_LABEL_VALUE_INPUT_META_PAUSE_TOGGLE, RETROK_UNKNOWN,
-      RARCH_PAUSE_TOGGLE, NO_BTN, NO_BTN, 0,
+      MENU_ENUM_LABEL_VALUE_INPUT_META_PAUSE_TOGGLE, RETROK_BACKSPACE,
+      RARCH_PAUSE_TOGGLE, NO_BTN, 5, 0,
       true
    },
    {
@@ -1058,15 +1058,15 @@ static const struct retro_keybind retro_keybinds_1[] = {
    {
       NULL, NULL,
       AXIS_NONE, AXIS_NONE,
-      MENU_ENUM_LABEL_VALUE_INPUT_META_LOAD_STATE_KEY, RETROK_UNKNOWN,
-      RARCH_LOAD_STATE_KEY, NO_BTN, NO_BTN, 0,
+      MENU_ENUM_LABEL_VALUE_INPUT_META_LOAD_STATE_KEY, RETROK_LCTRL,
+      RARCH_LOAD_STATE_KEY, NO_BTN, 2, 0,
       true
    },
    {
       NULL, NULL,
       AXIS_NONE, AXIS_NONE,
-      MENU_ENUM_LABEL_VALUE_INPUT_META_SAVE_STATE_KEY, RETROK_UNKNOWN,
-      RARCH_SAVE_STATE_KEY, NO_BTN, NO_BTN, 0,
+      MENU_ENUM_LABEL_VALUE_INPUT_META_SAVE_STATE_KEY, RETROK_LALT,
+      RARCH_SAVE_STATE_KEY, NO_BTN, 1, 0,
       true
    },
    {
@@ -1247,8 +1247,8 @@ static const struct retro_keybind retro_keybinds_1[] = {
    {
       NULL, NULL,
       AXIS_NONE, AXIS_NONE,
-      MENU_ENUM_LABEL_VALUE_INPUT_META_GAME_FOCUS_TOGGLE, RETROK_UNKNOWN,
-      RARCH_GAME_FOCUS_TOGGLE, NO_BTN, NO_BTN, 0,
+      MENU_ENUM_LABEL_VALUE_INPUT_META_GAME_FOCUS_TOGGLE, RETROK_PAGEDOWN,
+      RARCH_GAME_FOCUS_TOGGLE, NO_BTN, 7, 0,
       true
    },
    {
@@ -1289,8 +1289,8 @@ static const struct retro_keybind retro_keybinds_1[] = {
    {
       NULL, NULL,
       AXIS_NONE, AXIS_NONE,
-      MENU_ENUM_LABEL_VALUE_INPUT_META_FPS_TOGGLE, RETROK_UNKNOWN,
-      RARCH_FPS_TOGGLE, NO_BTN, NO_BTN, 0,
+      MENU_ENUM_LABEL_VALUE_INPUT_META_FPS_TOGGLE, RETROK_PAGEUP,
+      RARCH_FPS_TOGGLE, NO_BTN, 6, 0,
       true
    },
    {
-- 
2.45.2.windows.1

